src/folding/nova/nifs.rs

```diff
@@ -115,12 +133,27 @@ where
 
     // NIFS.V
     pub fn verify(
-        // r comes from the transcript, and is a n-bit (N_BITS_CHALLENGE) element
-        r: C::ScalarField,
+        poseidon_config: PoseidonConfig<C::ScalarField>,
```

Comment:
Here the idea was that `r` comes from the transcript which is generated outside (previous) to this method call.